src/components/routes/Account/FirebaseEmailHandler/FirebaseEmailHandler.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { parseQueryString } from 'utils/queryParams';
+import GeneralWrapper from 'shared/GeneralWrapper';
+import FirebaseEmailHandlerContainer from './FirebaseEmailHandler.container';
+import EmailVerify from './EmailVerify';
+import PasswordReset from './PasswordReset';
+
+interface State {
+  renderPasswordReset: boolean;
+  renderEmailVerify: boolean;
+  oobCode: string;
+}
+
+interface QueryParams {
+  mode?: string;
+  oobCode?: string;
+  continueUrl?: string;
+  lang?: string;
+}
+
+export default class FirebaseEmailHandler extends React.Component<RouterProps> {
+  readonly state: State = {
+    renderPasswordReset: false,
+    renderEmailVerify: false,
+    oobCode: '',
+  };
+
+  componentDidMount() {
+    const queryParams: QueryParams = parseQueryString(this.props.location.search);
+    const { mode, oobCode } = queryParams;
+
+    if (!mode) {
+      alert('Error, no mode defined.');
```

Comment:
wondering if these alerts are useful to the user, maybe we can make these a little more user-friendly?

Reply:
i think for this case, it's ok since there's nothing the user can do if they were given a faulty link.